connect/clients/nats.py

```diff
@@ -153,6 +182,102 @@ def nats_timing_event_handler(msg: Msg):
     )
 
 
+async def nats_retransmit_event_handler(msg: Msg):
+    """
+    Callback for NATS 'nats_retransmit_subject' messages
+    """
+    subject = msg.subject
+    reply = msg.reply
+    data = msg.data.decode()
+    logger.trace(
+        f"nats_retransmit_event_handler: received a message on {subject} {reply}"
+    )
+
+    message = json.loads(data)
+    await do_retransmit(message, -1)
+
+
+async def do_retransmit(message: dict, queue_pos: int):
```

Comment:
Would you please update the docstring to include a bit of info on the queue_pos parameter? I'm not entirely following how this is used or when it would be -1

Reply:
I'll answer here as well.  do_retransmit() can be called either from the NATS handler nats_retransmit_event_handler() that receives messages from the NATS server or from retransmitter() that is walking the queue of messages that still have failed retransmits.  If the message is coming directly from NATS, it won't be queued yet and it's helpful to know that, so we can queue it after another failure.  In the case where the message is already queued and we have a successful retransmit, we can delete the message from the queue and  it is again helpful to know the queue position.